lib/activerecord-bitemporal/bitemporal.rb

```diff
@@ -521,14 +522,25 @@ def scope_relation(record, relation)
           end
         }
 
-        bitemporal_scope = finder_class.unscoped.ignore_valid_datetime
+        valid_at_scope = finder_class.unscoped.ignore_valid_datetime
             .where("valid_from < ?", valid_to).where("? < valid_to", valid_from)
             .yield_self { |scope|
               # MEMO: #dup などでコピーした場合、id は存在しないが swapped_id のみ存在するケースがあるので
               # id と swapped_id の両方が存在する場合のみクエリを追加する
               record.id && record.swapped_id ? scope.where.not(id: record.swapped_id) : scope
             }
-        relation.merge(bitemporal_scope)
+
+        # MEMO: Must refer Time.current, when not new record
```

Comment:
Does this mean we cannot update a record that is not duplicated in a transaction? If it is not a new record, `created_at` will point to the current time, so I guess it will not pass the uniqueness check. Or will this check be ignored when updating?

Reply:
BTDM creates a new record when `#update`.
So, validation at `#update` refers to` Time.current` instead of `created_at`.